integration_tests/src/main/python/join_test.py

```diff
@@ -397,17 +397,44 @@ def do_join(spark):
         return left.join(broadcast(right), left.a > f.log(right.r_a), join_type)
     assert_gpu_and_cpu_are_equal_collect(do_join)
 
+@allow_non_gpu('BroadcastExchangeExec')
+@ignore_order(local=True)
+@pytest.mark.parametrize('data_gen', [IntegerGen(), LongGen(), pytest.param(FloatGen(), marks=[incompat]), pytest.param(DoubleGen(), marks=[incompat])], ids=idfn)
+@pytest.mark.parametrize('join_type', ['Cross', 'Right'], ids=idfn)
```

Comment:
Updated. Previously, it's mainly because we don't support other join types with left build. https://github.com/NVIDIA/spark-rapids/blob/25ffa873ad6211f14d5e402805e537b52d60d2dc/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastNestedLoopJoinExecBase.scala#L68 
Please see [here](https://github.com/NVIDIA/spark-rapids/pull/9635/files#diff-36070c90652877204943a4cbe235577d7464060469d87bf521dd65fe014f3413R410-R412) with updated test cases to cover more join types.